crates/rome_js_analyze/src/analyzers/nursery/no_fallthrough_switch_clause.rs

```diff
@@ -88,31 +206,82 @@ impl Rule for NoFallthroughSwitchClause {
     }
 }
 
-fn case_fell(case: &AnyJsSwitchClause) -> bool {
-    let statements = case.consequent();
-    !has_fall_blocker_statement(&statements) && statements.iter().count() != 0
-}
-
-fn has_fall_blocker_statement(statements: &JsStatementList) -> bool {
-    for statement in statements.iter() {
-        if is_fall_blocker_statement(&statement) {
-            return true;
-        }
-        if let Some(block_statement) = JsBlockStatement::cast_ref(statement.syntax()) {
-            if has_fall_blocker_statement(&block_statement.statements()) {
+fn has_switch_statement(control_flow_root: &JsSyntaxNode) -> bool {
+    let mut iter = control_flow_root.preorder();
+    while let Some(node) = iter.next() {
+        if let WalkEvent::Enter(node) = node {
+            if JsSwitchStatement::can_cast(node.kind()) {
                 return true;
+            } else if AnyJsControlFlowRoot::can_cast(node.kind()) && control_flow_root != &node {
+                iter.skip_subtree()
             }
         }
     }
     false
 }
 
-fn is_fall_blocker_statement(statement: &AnyJsStatement) -> bool {
-    matches!(
-        statement,
-        AnyJsStatement::JsBreakStatement(_)
-            | AnyJsStatement::JsReturnStatement(_)
-            | AnyJsStatement::JsThrowStatement(_)
-            | AnyJsStatement::JsContinueStatement(_)
-    )
+fn register_fallthrough_switch_clauses(
+    block_to_switch_clause_range: &FxHashMap<u32, Option<TextRange>>,
+    visited_blocks: &FxHashSet<u32>,
+    cfg: &rome_control_flow::ControlFlowGraph<JsLanguage>,
+    fallthrough: &mut Vec<TextRange>,
+) {
+    let mut current_switch_clause = None;
+    // Register all switch clauses as block to process.
+    let mut block_stack: Vec<u32> = block_to_switch_clause_range.keys().copied().collect();
```

Comment:
We wouldn't need to copy the keys if we pass `block_to_switch_clause_range` by value.

Reply:
I think not because we use `block_to_switch_clause_range` in this function.